### DISCUSSION TITLE
Ability to set default callbacks, failure_callbacks, failure_state, etc. at the Process level

### DISCUSSION THREAD

**Discussion by dwohlfahrt** [Borderless360/django-logic#66] [Ideas]
I have a process where all of the transitions have the same callbacks, failure_callbacks and failure_state.  Would be convenient if you could define these on the Process and use those as the defaults for all process transitions.  You could still override these for any given transition by explicitly specifying different ones.

**Comment by edigerad** [comment]
Thank you for the suggestion! You can achieve this by creating a custom Transition class with shared callbacks, failure_callbacks, and failure_state. This makes your code reusable and avoids repetition. Here's a simplified example for reference:


python
```
class ApproveDocument(Transition):
    def __init__(self, side_effects=None, callbacks=None):
        side_effects = side_effects or [actions.notify_approver]
        callbacks = callbacks or [actions.log_approval]
        super().__init__(
            action_name='approve_document',
            sources=['draft'],
            target='approved',
            side_effects=side_effects,
            callbacks=callbacks,
            failure_callbacks=[actions.handle_approval_failure]
        )
```
Using this approach, you can define shared logic once and reuse it across multiple transitions. Let me know if you need further clarification!
